optimade/server/routers/utils.py

```diff
@@ -144,8 +144,8 @@ def get_included_relationships(
             if entry_relationship is not None:
                 refs = entry_relationship.get("data", [])
                 for ref in refs:
-                    # could check here and raise a warning if any IDs clash
-                    endpoint_includes[entry_type][ref["id"]] = ref
+                    if ref["id"] not in endpoint_includes[entry_type]:
+                        endpoint_includes[entry_type][ref["id"]] = ref
```

Comment:
Trying to remember why I wrote the original comment, is it still possible to provide duplicate IDs in a relationship? Could happen accidentally, feel like it's the implementer rather than the user that needs to be warned though...

Reply:
Yeah. I was also a bit unsure about this one and the original intent of the comment.
I tried setting this up with a warning, but when I started writing tests for it. I could never write one that worked, because other things would fail if any two resources had the same `id`.
So I simply understood it instead to mean that you didn't want to continuously update and overwrite existing entries in the dictionary. This will fix that :)